treeherder/model/sql/jobs.json

```diff
@@ -98,13 +98,13 @@
         },
         "set_result_set":{
 
-            "sql":"INSERT INTO `result_set` (`author`, `revision_hash`,`push_timestamp`, `active_status`)
-                SELECT ?,?,?,?
+            "sql":"INSERT INTO `result_set` (`author`, `revision_hash`,`long_revision`,`short_revision`,`push_timestamp`, `active_status`)
+                SELECT ?,?,?,?,?,?
                 FROM DUAL
                 WHERE NOT EXISTS (
                     SELECT `revision_hash`, `push_timestamp`
                     FROM `result_set`
-                    WHERE `revision_hash` = ?
+                    WHERE `revision_hash` = ? or `long_revision` = ? or `short_revision` = ?
```

Comment:
I think the WHERE clause for this and the `set_revision` proc need to stay the same as they are, otherwise when we first deploy this we'll end up with dupes - since in the minutes leading up to the deploy we'll have inserts for rows that could then be inserted again, since they'll now have a `long_revision` / `short_revision`, whereas the previous row was NULL, if that makes sense?

At a later date, we can make the WHERE clause more specific.